Added verb synonyms

Gman98ish opened the pull request:
Closes issue #18 

I've done what you've suggested, so if you specify another verb it will use that instead.

I'm new to Clojure so I'm not sure on the implementation

apply-synonyms finds values in the item map, and copies the values from
synonyms over. E.g.
:unlock "Won't open" :open :unlock
Changes to
:unlock "Won't open" :open "Won't open"

Gman98ish committed:
Added an apply-synonyms method

apply-synonyms finds values in the item map, and copies the values from
synonyms over. E.g.
:unlock "Won't open" :open :unlock
Changes to
:unlock "Won't open" :open "Won't open"

facundoolano commented on src/advenjure/items.cljc:
```diff
@@ -29,16 +29,25 @@
+(defn apply-synonyms
+  [item]
+  (let [keys (keys item)]
+    (map (fn [field]
```
look into the `reduce` function, that's what you want to use here instead of `map`. You walk through each key building a new map: if the value is a key in the old map, use the value it points to, otherwise keep the current one. Something in the lines of `(assoc new-map key (get old-map value value))`.

BTW you don't want to be using stuff like `.indexOf`, clojure has great manipulation functions that you can take advantage on. [This might help](https://clojure.org/api/cheatsheet).

facundoolano commented:
Thanks for contributing! You came up with a much simpler idea than what I had planned: I didn't think about copying the value on item creation, I was thinking in traversing the map when it was read during the game.

I left a comment on the code. Also, it would be great if you can add a unit test for this.

Gman98ish committed:
Refactored apply synonyms function to use reduce and clojure data structures

I've also added a unit test to assert that verb synonyms are correctly
applied

Gman98ish commented:
Thanks for the feedback! It was really helpful. I've changed the function to use `reduce-kv`, and switched to `get` instead of `indexOf`. I've also added a unit test at the bottom of items_test.clj which just checks that synonyms are properly applied when you make an item.

facundoolano commented:
Thanks!
